### PR TITLE
Unify provider READMEs and add CSP Config examples for Upload

### PR DESCRIPTION
**Pull request by derrickmehaffy**
### What does it do?

Cleans up all the READMEs to fix their content and make all Email Providers / Upload Provider with the same structure
Added CSP config for upload providers

### Why is it needed?

Was using old package names, wrong URLs, and was missing config for the Security Middleware

### How to test it?

N/A

### Related issue(s)/PR(s)

N/A

### CODE REVIEW

File: packages/providers/email-amazon-ses/package.json [hunk 1]
```diff
@@ -42,5 +42,8 @@
   "engines": {
     "node": ">=12.x.x <=16.x.x",
     "npm": ">=6.0.0"
+  },
+  "strapi": {
+    "isProvider": true
```
Comment (petersg83) [inline packages/providers/email-amazon-ses/package.json:47]: Actually we don't use this key at all. I suggest to remove it for the moment.
The current logic is 
```
strapi: {
   kind: 'plugin'
}
```
We could use the same for providers: `kind: 'provider'` but that would need a bit of thinking first. `provider` or `email-provider` or else?

I propose to remove the key from all packages. wdyt @alexandrebodin ?

File: packages/providers/email-nodemailer/.editorconfig [hunk 2]
```diff
@@ -0,0 +1,16 @@
+root = true
```
Comment (petersg83) [inline packages/providers/email-nodemailer/.editorconfig:1]: There is an `.editorconfig` file at the root of the project, it isn't enough?

File: packages/providers/email-nodemailer/.gitignore [hunk 3]
```diff
@@ -0,0 +1,96 @@
+############################
```
Comment (petersg83) [inline packages/providers/email-nodemailer/.gitignore:1]: Same question

File: packages/providers/upload-aws-s3/README.md [hunk 4]
```diff
@@ -30,26 +51,42 @@ module.exports = ({ env }) => ({
 });
 ```
 
+### Security Middleware Configuration
+
+Due to the default settings in the Strapi Security Middleware you will need to modify the `contentSecurityPolicy` settings to properly see thumbnail previews in the Media Library. You should replace `strapi::security` string with the object instead as explained in the [middleware configuration](https://docs.strapi.io/developer-docs/latest/setup-deployment-guides/configurations/required/middlewares.html#loading-order) documentation.
```
Comment (petersg83) [inline packages/providers/upload-aws-s3/README.md:56]: I propose 
```diff
- with the object instead
+ with the object bellow instead
```

File: packages/providers/upload-rackspace/README.md [hunk 5]
```diff
@@ -28,12 +49,29 @@ module.exports = ({ env }) => ({
...
+  // ...
+  {
+    name: 'strapi::security',
+    config: {
+      contentSecurityPolicy: {
+        useDefaults: true,
+        directives: {
+          'connect-src': ["'self'", 'https:'],
+          'img-src': ["'self'", 'data:', 'blob:', 'storage.clouddrive.com'],
+          'media-src': ["'self'", 'data:', 'blob:', 'storage.clouddrive.com'],
```
Comment (petersg83) [inline packages/providers/upload-rackspace/README.md:69]: so you are sure it is this url?

**Review by petersg83 — CHANGES_REQUESTED** [review]
Nice update of the READMEs !

File: packages/providers/email-nodemailer/.editorconfig (hunk 2 above)
Comment (derrickmehaffy) [reply packages/providers/email-nodemailer/.editorconfig:1]: All the other providers had it so I just mirrored the same

File: packages/providers/email-nodemailer/.gitignore (hunk 3 above)
Comment (derrickmehaffy) [reply packages/providers/email-nodemailer/.gitignore:1]: same answer

File: packages/providers/email-nodemailer/.editorconfig (hunk 2 above)
Comment (petersg83) [reply packages/providers/email-nodemailer/.editorconfig:1]: Ok, I'll make a PR to remove them all I think and check if @alexandrebodin has an objection. You can let this file and I'll remove it in my PR

**Review by petersg83 — APPROVED** [review]
LGTM

**Comment by derrickmehaffy** [comment]
This pull request has been mentioned on **Strapi Community Forum**. There might be relevant details there:

https://forum.strapi.io/t/how-to-configure-internal-middlewares/13125/2

**Pull request merged**